CMakeLists.txt

```diff
@@ -0,0 +1,658 @@
+                      "Passing \"-DWITH_BDB=OFF\" will suppress this warning."
+      )
+    endif()
+  endif()
+endif()
+cmake_dependent_option(BUILD_WALLET_TOOL "Build bitcoin-wallet tool." ${BUILD_TESTS} "ENABLE_WALLET" OFF)
+
+option(ENABLE_HARDENING "Attempt to harden the resulting executables." ON)
+option(REDUCE_EXPORTS "Attempt to reduce exported symbols in the resulting executables." OFF)
+option(WERROR "Treat compiler warnings as errors." OFF)
+option(WITH_CCACHE "Attempt to use ccache for compiling." ON)
+
+option(WITH_NATPMP "Enable NAT-PMP." OFF)
+if(WITH_NATPMP)
+  find_package(NATPMP MODULE REQUIRED)
+endif()
+
+option(WITH_MINIUPNPC "Enable UPnP." OFF)
+if(WITH_MINIUPNPC)
+  find_package(MiniUPnPc MODULE REQUIRED)
+endif()
+
+option(WITH_ZMQ "Enable ZMQ notifications." OFF)
+if(WITH_ZMQ)
+  if(VCPKG_TARGET_TRIPLET)
+    find_package(ZeroMQ CONFIG REQUIRED)
+  else()
+    # The ZeroMQ project has provided config files since v4.2.2.
+    # TODO: Switch to find_package(ZeroMQ) at some point in the future.
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(libzmq REQUIRED IMPORTED_TARGET libzmq>=4)
+    # TODO: This command will be redundant once
+    #       https://github.com/bitcoin/bitcoin/pull/30508 is merged.
+    target_link_libraries(PkgConfig::libzmq INTERFACE
+      $<$<PLATFORM_ID:Windows>:iphlpapi;ws2_32>
+    )
+  endif()
+endif()
+
+option(WITH_USDT "Enable tracepoints for Userspace, Statically Defined Tracing." OFF)
+if(WITH_USDT)
+  find_package(USDT MODULE REQUIRED)
+endif()
+
+cmake_dependent_option(ENABLE_EXTERNAL_SIGNER "Enable external signer support." ON "NOT WIN32" OFF)
+
+cmake_dependent_option(WITH_QRENCODE "Enable QR code support." ON "BUILD_GUI" OFF)
+if(WITH_QRENCODE)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(libqrencode REQUIRED IMPORTED_TARGET libqrencode)
+  set(USE_QRCODE TRUE)
+endif()
+
+cmake_dependent_option(WITH_DBUS "Enable DBus support." ON "CMAKE_SYSTEM_NAME STREQUAL \"Linux\" AND BUILD_GUI" OFF)
+
+option(WITH_MULTIPROCESS "Build multiprocess bitcoin-node and bitcoin-gui executables in addition to monolithic bitcoind and bitcoin-qt executables. Requires libmultiprocess library. Experimental." OFF)
+if(WITH_MULTIPROCESS)
+  find_package(Libmultiprocess COMPONENTS Lib)
+  find_package(LibmultiprocessNative COMPONENTS Bin
+    NAMES Libmultiprocess
+  )
+endif()
+
+cmake_dependent_option(BUILD_GUI_TESTS "Build test_bitcoin-qt executable." ON "BUILD_GUI;BUILD_TESTS" OFF)
+if(BUILD_GUI)
+  set(qt_components Core Gui Widgets LinguistTools)
+  if(ENABLE_WALLET)
+    list(APPEND qt_components Network)
+  endif()
+  if(WITH_DBUS)
+    list(APPEND qt_components DBus)
+    set(USE_DBUS TRUE)
+  endif()
+  if(BUILD_GUI_TESTS)
+    list(APPEND qt_components Test)
+  endif()
+  find_package(Qt5 5.11.3 MODULE REQUIRED
+    COMPONENTS ${qt_components}
+  )
+  unset(qt_components)
+endif()
+
+option(BUILD_BENCH "Build bench_bitcoin executable." OFF)
```

Comment:
Previously benchmarks were [enabled by default](https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L180), right?

Is this why the build is so much faster now?

I can run the benchmarks from the command line, but the IDE is complaining they're not part of the project. Is that expected?
<img src="https://github.com/user-attachments/assets/e4f0d837-bc70-48f6-8a75-e24b45cdeb42">

Reply:
> Previously benchmarks were [enabled by default](https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L180), right?

Right.

> Is this why the build is so much faster now?

As [mentioned](https://github.com/bitcoin/bitcoin/pull/30454#issuecomment-2251181234) earlier:
> Regarding the part about "the default with autoconf is on" part, in the CMake working group, we decided to enable by default only the "essential" build targets that most developers build most often in their everyday work. This means, for instance, that `-DBUILD_BENCH` and `-DBUILD_GUI` are `OFF` by default.

> I can run the benchmarks from the command line, but the IDE is complaining they're not part of the project. Is that expected?

Is the project configured with `-DBUILD_BENCH=ON`?

Reply:
yeah (even edited the scripts to have them `ON` by default), I can run a benchmark from the terminal, but the IDE doesn't see them as part of the project.
Maybe @rot13maxi can confirm or deny whether CLion sees the benchmarks.

Reply:
@paplorinc 

Please open an issue in https://github.com/hebasto/bitcoin/issues with detailed steps to reproduce the problem from the scratch.

Reply:
I managed to make it work, I think the internal IDE setting were overwriting the ones I was setting from the command line.

For the record, with these settings the bench and fuzz projects are enabled (though no IDE assistance - i.e. run from the IDE -, but that's a different issue):
<img src="https://github.com/user-attachments/assets/06b10237-c4b3-4ca3-8547-b3299d214070">

I did open an unrelated suggestion for better error messages: https://github.com/hebasto/bitcoin/issues/306

Reply:
> I managed to make it work...

I consider this conversation as resolved then.